eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Bbody.hs

```diff
@@ -308,3 +322,18 @@ instance
   where
   wrapFailed = LedgersFailure
   wrapEvent = ShelleyInAlonzoEvent . LedgersEvent
+
+totalRefScriptSizeInBlock ::
+  (AlonzoEraTx era, BabbageEraTxBody era) => ProtVer -> StrictSeq (Tx era) -> UTxO era -> Int
+totalRefScriptSizeInBlock protVer txs (UTxO utxo)
+  | pvMajor protVer <= natVersion @10 =
+      getSum $ foldMap' (Monoid.Sum . txNonDistinctRefScriptsSize (UTxO utxo)) txs
+  | otherwise =
+      snd $ F.foldl' accum (utxo, 0) txs
+  where
+    accum (!accUtxo, !accSum) tx =
+      let updatedUtxo = accUtxo `Map.union` unUTxO toAdd
+          toAdd
+            | IsValid True <- tx ^. isValidTxL = txouts $ tx ^. bodyTxL
+            | otherwise = collOuts $ tx ^. bodyTxL
+       in (updatedUtxo, accSum + txNonDistinctRefScriptsSize (UTxO updatedUtxo) tx)
```

Comment:
We should not take into consideration outputs produced by the transaction, while computing usage of refscripts for that transaction.
```suggestion
       in (updatedUtxo, accSum + txNonDistinctRefScriptsSize accUtxo tx)
```
This wouldn't be the end of the world, since such usage would result in another predicate failure, but it is better to implement it in such a way that is not only accurate, but also conveys the intent correctly.